hack/log/log-dump.sh

```diff
@@ -127,8 +127,52 @@ dump_workload_cluster_logs() {
     wait
 }
 
+dump_workload_cluster_logs_windows() {
+    echo "Deploying log-dump-daemonset-windows"
+    "${KUBECTL}" apply -f "${REPO_ROOT}/hack/log/log-dump-daemonset-windows.yaml"
+    echo "Waiting for log-dump-daemonset-windows"
+    "${KUBECTL}" wait pod -l app=log-dump-node-windows --for=condition=Ready --timeout=5m
```

Comment:
Does anyone know if this would cause issues if there aren't any windows nodes / the DS doesn't schedule any pods?
I'm assuming not but will try and test this out...

Reply:
I think it should be fine since total number of pods will be 0. Did you try it out?

Reply:
I tried this out and the kubectl command returned with exitcode 1 (and `error: no matches resources found`).

when I ran  `kubectl get pod -l app=log-dump-node-windows -ojsonpath='{.items[*].metadata.name}'` like on the next line that returns success and no output.

Should we let the wait command fail and exit out of the function or should I add `|| true` there and just the for loop no-op?

Reply:
I added a check to see if TEST_WINDOWS envvar was set and only call dump_workload_cluster_logs_windows if it is